### PR TITLE
Minimum jump to reach end dynamic programming

### PR DESCRIPTION
**Pull request by hieetesh**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by hieetesh** [comment]
I am facing Pre-commit failure, my code is working fine. I am not able to understand the issue. Can anyone help?

**Comment by hieetesh** [comment]
Hey @Kush1101, I have tested my code multiple times, its working fine. I am not able to understand this PR commit failure. Will you please help. Thanks in advance :)

**Comment by hieetesh** [comment]
I have passed all the tests :)

**Comment by anneCoder1805** [comment]
> I have passed all the tests :)

Can you please let me know whta you did to overcome the issue? I, like many others am facing the same problem.

**Comment by hieetesh** [comment]
Sure @anneCoder1805. Actually we have to test our code with "black" and "flake8" to make it properly indentated and to remove extra white spaces. Install black using: "pip3 install black" and then execute it using:  "black filename". Same to do with flake8: pip3 install flake8 and flake8 filename. Flake8 will give you all the necessary updates you need to do to make your code pass the tests. (For example: Remove whitespace at line 17 or Line 18 is longer than 83 characters).

**Comment by anneCoder1805** [comment]
> Sure @anneCoder1805. Actually we have to test our code with "black" and "flake8" to make it properly indentated and to remove extra white spaces. Install black using: "pip3 install black" and then execute it using: "black filename". Same to do with flake8: pip3 install flake8 and flake8 filename. Flake8 will give you all the necessary updates you need to do to make your code pass the tests. (For example: Remove whitespace at line 17 or Line 18 is longer than 83 characters).

Thanks a lot.

**Comment by hieetesh** [comment]
Hello @Kush1101, will you please review and accept my pull request. Thank you in advance :)

**Review by 53jk1 — APPROVED** [review]

**Comment by hieetesh** [comment]
@53jk1, will you please merge my pull request. Thank you in advance :)

File: dynamic_programming/min_jump_to_reach_end.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            if i >= maxReach:
+                return -1
+
+            # re-initialize the steps to the amount
+            # of steps to reach maxReach from position i.
+            step = maxReach - i
+    return -1
+
+
+if __name__ == "__main__":
```
Comment (poyea) [inline dynamic_programming/min_jump_to_reach_end.py:64]:
```suggestion
if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

File: dynamic_programming/min_jump_to_reach_end.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            Input:  arr[] = {1, 3, 5, 8, 9, 2, 6, 7, 6, 8, 9}
+            Output: 3 (1-> 3 -> 8 -> 9)
+"""
+
+
+def min_jump_to_reach_end(arr: list) -> int:
+    """
+    >>> min_jump_to_reach_end([1, 3, 5, 8, 9, 2, 6, 7, 6, 8, 9])
+    3
+    """
```
Comment (poyea) [inline dynamic_programming/min_jump_to_reach_end.py:15]: How about two to three more tests for `n <= 1` and `arr[0] == 0`?

**Comment by hieetesh** [comment]
@poyea I have added your suggested test cases. Please approve my pull request. Thanks in advance :)

File: dynamic_programming/min_jump_to_reach_end.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+    arr = [1, 3, 5, 8, 9, 2, 6, 7, 6, 8, 9]
+    print(min_jump_to_reach_end(arr))
+    print(min_jump_to_reach_end([]))
+    print(min_jump_to_reach_end([0, 1, 10, 15, 2, 2]))
```
Comment (poyea) [inline dynamic_programming/min_jump_to_reach_end.py:71]: What I mean was put the test cases in the doctest. Like what you've done like:
```   
    """
    >>> min_jump_to_reach_end([1, 3, 5, 8, 9, 2, 6, 7, 6, 8, 9])
    3
    """

**Comment by hieetesh** [comment]
@poyea, Please check now

**Comment by hieetesh** [comment]
@poyea , Please merge my pull request if everything looks good to you. Thank you :)

**Pull request closed without merging**